feat(inventory): enforce account types on groups

jniles opened the pull request:
Enforce the account types for each type of account on the inventory group via the `bhAccountSelect` component.

Closes #4703.

Here is what it looks like:
![u95E40ge8u](https://user-images.githubusercontent.com/896472/89422300-0ecc6d80-d72d-11ea-8389-a2f9b2b7b391.gif)

jniles committed:
feat(inventory): enforce account types on groups

Enforce the account types for each type of account on the inventory
group via the bhAccountSelect component.

Closes #4703

mbayopanda commented on client/src/js/components/bhAccountSelect.js:
```diff
@@ -44,16 +46,39 @@ function AccountSelectController(Accounts, FormatTreeData, bhConstants, $scope)
+      parsed = [types];
+    } else {
+      throw new Error('Cannot parse account types from '.concat(types));
+    }
```
I guess we are not waiting to receive an object if it can occur maybe we can handle that also here

jniles replied:
I actually think it's a good thing to throw an error if the developer doesn't respect the API.  I can't imagine a scenario in which an object should be accepted.

mbayopanda commented on client/src/modules/inventory/configuration/groups/modals/actions.tmpl.html:
```diff
@@ -41,6 +41,7 @@
       data-sales-account
       name="sales_account"
       account-id="$ctrl.session.sales_account"
+      account-type-id="$ctrl.INCOME_ACCOUNT_TYPE_ID"
```
👍

mbayopanda reviewed (approved):
LGTM 👍

jmcameron reviewed (approved):
I tested this (with Bhima_test) and it worked fine.  I looked through the code and looked okay too.  
I was curious about two things:

- Why the Bhima Constants involved are INCOME/ASSET/EXPENSE yet the names used for these (in English) on the Add Inventory Group page are SALE/STOCK/COST?
- Why the 1x accounts (Durable Resources) are not included on this form.  Are the buildings not considered inventory?

jniles committed:
fix(accounts): use correct $bhValue declaration

Uses the correct implementation of $bhValue.  Also makes the
bhAccountSelect component use the correct controller!

jniles committed:
test: update accents in inventory data

Makes sure all "medicaments" are properly accented in our test dataset
so our tests can select the correct elements.

mbayopanda commented on client/src/modules/templates/bhAccountSelect.tmpl.html:
```diff
@@ -1,4 +1,4 @@
-<div ng-form="AccountForm" bh-account-select ng-model-options="{ updateOn: 'default' }">
+<div ng-form="{{ $ctrl.name }}" bh-account-select ng-model-options="{ updateOn: 'default' }">
```
👍

mbayopanda reviewed (approved):
The behavior expected is working fine. LGTM 👍
